test/unit/puppet_fact_parser_test.rb

```diff
@@ -233,8 +234,35 @@ def setup
     assert_equal '172.30.43.0', parser.interfaces['local_area_connection_2'][:network]
   end
 
+  test "should create connection between new os and new arch on facts parsing" do
+    assert os_to_arch_connectivity_check
+  end
+
+  test "should create connection between existing os and new arch on facts parsing" do
+    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
+    assert os_to_arch_connectivity_check
```

Comment:
Don't you think it will be to specific to do the connection of those two
facts over there?
By the way thanks for all the good comments, taken care of all but this one.


2015-06-04 17:41 GMT+03:00 Dominic Cleal <notifications@github.com>:

> In test/unit/puppet_fact_parser_test.rb
> <https://github.com/theforeman/foreman/pull/2417#discussion_r31726531>:
>
> > @@ -233,8 +234,35 @@ def setup
> >      assert_equal '172.30.43.0', parser.interfaces['local_area_connection_2'][:network]
> >    end
> >
> > +  test "should create connection between new os and new arch on facts parsing" do
> > +    assert os_to_arch_connectivity_check
> > +  end
> > +
> > +  test "should create connection between existing os and new arch on facts parsing" do
> > +    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
> > +    assert os_to_arch_connectivity_check
>
> Good point! Perhaps just call both methods in the test.. or this points to
> an issue that we should actually be doing this in Host::Base or
> Host::Managed, in the populate_fields_from_facts method. An advantage of
> doing it there would be that this would equally apply to other parsers,
> such as Chef or Salt. @ares <https://github.com/ares>, what do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/theforeman/foreman/pull/2417/files#r31726531>.
>